### PR TITLE
Add intro section to LRE docs

### PR DESCRIPTION
**Pull request by natestemen**
## Description

Adding "beginner" friendly docs to LRE user guide.

### CODE REVIEW

File: docs/source/guide/lre-1-intro.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+noise_scaled_circuits = multivariate_layer_scaling(circuit, degree, fold_multiplier)
+num_scaled_circuits = len(noise_scaled_circuits)
+
+print(f"total number of noise-scaled circuits for LRE = {num_scaled_circuits}")
+print(
+    f"Average circuit depth = {sum(len(circuit) for circuit in noise_scaled_circuits) / num_scaled_circuits}"
+)
+```
+
+Even though though our circuit of interest has depth 8, the noise scaled circuits are on average much longer.
```
Comment (purva-thakre) [inline docs/source/guide/lre-1-intro.md:107]: > Even though though our circuit of interest has depth 8

I wouldn't state this explicitly. Randomized benchmarking circuits function doesn't generate a circuit of the same depth.

File: docs/source/guide/lre-1-intro.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    jupytext_version: 1.11.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# How do I use LRE?
+
+LRE works in two main steps: generate noise-scaled circuits, and apply inference to results from executed circuits.
```
Comment (purva-thakre) [inline docs/source/guide/lre-1-intro.md:16]:
```suggestion
LRE works in two main steps: generate layered noise-scaled circuits in a specific pattern, and apply inference to results from executed circuits.
```

File: docs/source/guide/lre-1-intro.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+```{code-cell} ipython3
+noisy = execute(circuit)
+ideal = execute(circuit, noise_level=0.0)
+print(f"Error without mitigation: {abs(ideal - noisy) :.5f}")
+```
+
+## Apply LRE directly
+
+With the circuit, and executor defined, we just need to choose the polynomial extrapolation degree as well as the fold multiplier.
```
Comment (purva-thakre) [inline docs/source/guide/lre-1-intro.md:65]:
```suggestion
With the circuit and executor defined, we just need to choose the polynomial extrapolation degree as well as the fold multiplier.
```

File: docs/source/guide/lre-1-intro.md (hunk 1 above)
Comment (natestemen) [reply docs/source/guide/lre-1-intro.md]: Good catch! Reworded in https://github.com/unitaryfund/mitiq/pull/2535/commits/ea14e692b5b93250de7ca9b062e8c6e56726b624.

File: docs/source/guide/lre-1-intro.md (hunk 2 above)
Comment (natestemen) [reply docs/source/guide/lre-1-intro.md]: I added a few more details in https://github.com/unitaryfund/mitiq/pull/2535/commits/f2f38290f13995f61aa2b555ea47ad94f3845eac.

**Pull request merged**